src/classes/CanvasRenderingContext2D.js

```diff
@@ -1053,8 +1053,16 @@ export default class CanvasRenderingContext2D {
     return this._fontStack[this._stackIndex];
   }
 
-  getImageData() {
-    return new ImageData(this._canvas.width, this._canvas.height);
+  getImageData(x, y, w, h) {
+    if (arguments.length < 4)
+      throw new TypeError(
```

Comment:
it will throw error in real api？

Reply:
Yes. You can try it in the browser console:
```
document.createElement("canvas").getContext("2d").getImageData()
```
(tested in Firefox, Chrome, Edge)

However, I will add a test for width/height parameters, method will throw if either is zero.